### DISCUSSION TITLE
Ship RBS type signatures with all of the components?

### DISCUSSION THREAD

**Discussion by nogweii** [piotrmurach/tty#86] [Ideas]
I've recently started a new project that pulls in a couple of components (Pastel and Prompt) to build a CLI tool to help manage my MTG card collection. 🎴 

As it's a new project in 2023, I thought it would be cool to explore how well type signatures work in the Ruby community. Unfortunately, there is not a lot of coverage. Thus me asking - would you consider including them in your gems?

If you don't want to maintain them manually, I've found that [sord](https://github.com/AaronC81/sord) produces pretty good signatures from YARD tags, which limits the amount of work. Including the generated file into the repo, or generating it on the fly as part of the upload process, would be pretty cool.

Thoughts?

**Comment by piotrmurach** [comment]
Hi @nogweii 👋 

For starters, thanks for using tty gems.

Short answer - no, I won't be adding type signatures files. Though I reserve the right to change my mind in the future 😄 

Why? I care for the size of the gems to be minimal. Type signatures files would increase package size - a killer for command line gems if you ask me. Secondly, I'm not super ethused about static typing in Ruby. I like Ruby for what it is. I'm also pragmatic and there is a lot of maintenance work and gems to release. My time is finite and limited so I want to spend it on developing features rather than maintaning type signatures.

I've tried `sord` with some of my gems and the resutls were good but not perfect. There were many errors, for example, sord seemed to struggle handling hash types among other things.

Thanks again for contributing your idea. 🙏
